controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,883 @@
+		}
+	}
+	clientId, clientSecret, err := getSecrets(h.secretsCache, config.Spec.AzureCredentialSecret)
+	if err != nil {
+		return fmt.Errorf("could not get secrets")
+	}
+	if clientId == "" || clientSecret == "" {
+		return fmt.Errorf(cannotBeNilError, "clientId and clientSecret", config.ClusterName)
+	}
+
+	azureAuthorizer, err := newClientAuthorizer(h.secretsCache, config.Spec)
+	if err != nil {
+		return err
+	}
+	serviceClient, err := newServiceClient(azureAuthorizer, config.Spec)
+	if err != nil {
+		return err
+	}
+	listOperators, err := serviceClient.ListOrchestrators(ctx, config.Spec.ResourceLocation, "managedClusters")
+	if err != nil {
+		return err
+	}
+	objOrchestrators := *listOperators.Orchestrators
+	validKubernetesVersion := false
+	for _, obj := range objOrchestrators {
+		if config.Spec.KubernetesVersion == to.String(obj.OrchestratorVersion) {
+			validKubernetesVersion = true
+			break
+		}
+	}
+	if !validKubernetesVersion {
+		return fmt.Errorf("field kubernetesVersion is invalid for non-import [%s] cluster location", config.ClusterName)
+	}
+	if hasAgentPoolProfile(config.Spec) {
+		for _, np := range config.Spec.NodePools {
+			if np.Name == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Name", config.ClusterName)
+			}
+			if np.Count == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Count", config.ClusterName)
+			}
+			if np.MaxPods == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.MaxPods", config.ClusterName)
+			}
+			if np.VMSize == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.VMSize", config.ClusterName)
+			}
+			if np.OsDiskSizeGB == 0 {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsDiskSizeGB", config.ClusterName)
+			}
+			if np.OsDiskType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OSDiskType", config.ClusterName)
+			}
+			if np.Mode == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.Mode", config.ClusterName)
+			}
+			if np.OsType == "" {
+				return fmt.Errorf(cannotBeNilError, "NodePool.OsType", config.ClusterName)
+			}
+		}
+	}
+	if (config.Spec.PublicAccess && config.Spec.PrivateAccess) || (!config.Spec.PublicAccess && !config.Spec.PrivateAccess) {
+		return fmt.Errorf("only one field PublicAccess or PrivateAccess have to be set for [%s] cluster", config.ClusterName)
+	}
+
+	if config.Spec.NetworkPolicy != "" {
+		networkPolicy := config.Spec.NetworkPolicy
+		if networkPolicy != string(containersv211.NetworkPolicyAzure) && networkPolicy != string(containersv211.NetworkPolicyCalico) {
+			return fmt.Errorf("wrong network policy value for [%s] cluster", config.ClusterName)
+		}
+	}
+	return nil
+}
+
+// createCASecret creates a secret containing ca and endpoint. These can be used to create a kubeconfig via
+// the go sdk
+func (h *Handler) createCASecret(ctx context.Context, config *aksv1.AKSClusterConfig) error {
+	kubeConfig, err := GetClusterKubeConfig(ctx, h.secretsCache, config.Spec)
```

Comment:
This seems like a weird way to get the endpoint and caData, is it not on the cluster object ?

Reply:
Yes, I was trying to get it in simpler way, it seems that `AccessProfile` object offers only `KubeConfig` to get this data

Reply:
Ok, I will rework this part